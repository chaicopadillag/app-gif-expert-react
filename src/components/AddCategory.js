import React, { useState } from 'react';
import PropTypes from 'prop-types';
const AddCategory = ({ setCategorias }) => {
	const [inputCategoria, setInputCategoria] = useState('');
	const handleCateogriaChange = (e) => {
		setInputCategoria(e.target.value);
	};
	const handleSubmit = (e) => {
		e.preventDefault();
		if (inputCategoria !== '') {
			setCategorias((cats) => [inputCategoria, ...cats]);
			setInputCategoria('');
		} else {
			alert('Ingrese una categor√≠a');
		}
	};
	return (
		<form method="POST" onSubmit={handleSubmit}>
			<input type="text" onChange={handleCateogriaChange} value={inputCategoria} placeholder="Buscar..." />
		</form>
	);
};
AddCategory.propTypes = {
	setCategorias: PropTypes.func.isRequired,
};
export default AddCategory;
