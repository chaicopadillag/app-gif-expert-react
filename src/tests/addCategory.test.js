import { shallow } from 'enzyme';
import AddCategory from '../components/AddCategory';

describe('Test add CategorÃ­a', () => {
	const setCategorias = jest.fn();
	let wraper = shallow(<AddCategory setCategorias={setCategorias} />);

	beforeEach(() => {
		jest.clearAllMocks();
		wraper = shallow(<AddCategory setCategorias={setCategorias} />);
	});

	test('Debe mostrar el Componente <AddCategory/>', () => {
		expect(wraper).toMatchSnapshot();
	});

	test('cambio de input', () => {
		const inputText = wraper.find('input'),
			value = 'Hola Mundo!';
		inputText.simulate('change', { target: { value } });
		expect(wraper.find('input').prop('value')).toBe(value);
	});

	test('El formulario no enviar datos', () => {
		wraper.find('form').simulate('submit', { preventDefault() {} });
		expect(setCategorias).not.toHaveBeenCalled();
	});

	test('debe de llamar el setCategorias y limpiar el input text', () => {
		const inputText = wraper.find('input'),
			value = 'Hola Mundo!';
		inputText.simulate('change', { target: { value } });
		wraper.find('form').simulate('submit', { preventDefault() {} });
		expect(setCategorias).toHaveBeenCalled();
		expect(setCategorias).toHaveBeenCalledTimes(1);
		expect(setCategorias).toHaveBeenCalledWith(expect.any(Function));

		expect(wraper.find('input').prop('value')).toBe('');
	});
});
