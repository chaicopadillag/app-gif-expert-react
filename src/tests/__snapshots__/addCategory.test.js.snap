// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test add Categor√≠a Debe mostrar el Componente <AddCategory/> 1`] = `
<form
  method="POST"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Buscar..."
    type="text"
    value=""
  />
</form>
`;
